### ISSUE TITLE
Update default visibility settings for variation options

### ISSUE THREAD

**Issue by jarekmorawski** [woothemes/woocommerce#40198]
We currently hide all newly added variation options from product details, which can be confusing to users because of the hidden icon we show in the list. I think there's no need to hide attributes used as variation options from product details and we can leave the choice to the user, thus making the visibility indicator more useful.

<img width="337" alt="image" src="https://github.com/woocommerce/woocommerce/assets/79307566/9c8edaca-b0f1-4312-afa0-ae841d2cc27a">

**Acceptance criteria**

- [x] All variation options have the `Show in product details` box unchecked by default.
- [x] We replace the 'hidden' icon with a new 'visible' icon (AkVNGImLgSqCObTQ3idVn7-fi-2780_272991). Make sure not to include the question mark.
- [x] The 'visible' icon is only shown when the `Show in product details` box in the variation option is checked.
- [x] The tooltip displayed when the user hovers over the 'hidden' icon is updated to: `Visible in product details`.
- [x] On hover, the cursor changes to `help`.

**Comment by mattsherman** [comment]
> All variation options are now displayed in product details by default

This can be confusing on a product page, as it will show up looking like this under "Additional information" when you have a specific variation selected (say, Small Red)...

```
Size: Small, Medium, Large
Color: Red, Green, Blue
```

I don't know if this is a good default.

**Comment by mdperez86** [comment]
@jarekmorawski could you please answer @mattsherman 's comment we need it to groom this issue

**Comment by jarekmorawski** [comment]
> I don't know if this is a good default.

I really wanted to align with how it's handled in the current UX, but I just checked it on a more robust product and it's bad.

<img width="1151" alt="image" src="https://github.com/woocommerce/woocommerce/assets/79307566/0046e643-90e0-4fc6-9169-2d2db48de81c">

I think the reverse behavior could work well, meaning that we'd let users _show_ the variation options in the `Additional information` tab if they really need it. I updated the ACs. cc @mattsherman
